_posts/2020-07-06-whats-so-super-about-super-tdma.md

```diff
@@ -0,0 +1,158 @@
+///     37(4):645-658, 2012.
+///
+///////////////////////////////////////////////////////////////////////////////
+
+import org.arl.fjage.*
+import org.arl.unet.*
+import org.arl.unet.phy.*
+import static org.arl.unet.Services.*
+
+///////////////////////////////////////////////////////////////////////////////
+// settings
+
+def slot = 422.ms               // default packet length is about 345 ms
+def range = 650.m               // about slot x 1540 m/s
+def time = 15.minutes           // simulation time
+def schedule = [[2, 3, 0, 0],   // schedule from [1]
+                [0, 0, 1, 3],
+                [0, 1, 0, 2]]
+
+///////////////////////////////////////////////////////////////////////////////
+// display documentation
+
+println """
+Equilateral triangle network
+----------------------------
+Internode distance:     ${range} m
+Slot length:            ${(1000*slot).round()} ms
+Simulation time:        ${time} s"""
+
+///////////////////////////////////////////////////////////////////////////////
+// simulate schedule
+
+simulate time, {
+
+  def n = []
+  n << node('1', address: 1, location: [0, 0, 0])
+  n << node('2', address: 2, location: [range, 0, 0])
+  n << node('3', address: 3, location: [0.5*range, 0.866*range, 0])
+
+  n.eachWithIndex { n1, i ->
+    n1.startup = {
+      def phy = agentForService PHYSICAL
+      phy[Physical.DATA].frameLength = phy[Physical.CONTROL].frameLength
+      add new TickerBehavior(1000*slot, {
+        def slen = schedule[i].size()
+        def s = schedule[i][(tickCount-1)%slen]
+        if (s) phy << new TxFrameReq(to: s, type: Physical.DATA)
+      })
+    }
+  }
+
+}
+
+// display statistics
+println """TX:                     ${trace.txCount}
+RX:                     ${trace.rxCount}
+Offered load:           ${trace.offeredLoad.round(3)}
+Throughput:             ${trace.throughput.round(3)}"""
+```
+
+First we set the distance between the nodes to be 650 m. With sound speed in water to be approximately 1540 m/s, the slot length turns out to be 422 ms which is set as can be seen in the code. The simulation time is set to 15 minutes and the high-throughput schedule is given as an input. The rest is pretty simple. The three network nodes are deployed at appropriate coordinates to form an equilateral triangle. The frame duration of the DATA channel packet is set to be smaller than the time slot length, i.e., we need to make sure that the frame length and data rate is such that the packet duration is smaller than 422 ms. Once we set this up, the packets are transmitted from each node in a `TickerBehavior` as per the schedule. Now, let's run this and look at the result:
+
+```term
+Equilateral triangle network
+----------------------------
+Internode distance:     650 m
+Slot length:            422 ms
+Simulation time:        900 s
+TX:                     3198
+RX:                     3011
+Offered load:           1.375
+Throughput:             1.295
```

Comment:
Might want some brief explanation on how to interpret `throughput`, and what throughput > 1 means (this is what we call _super_).

Reply:
Agreed. I have added much more details on this part now.